beacon_node/client/src/builder.rs

```diff
@@ -277,8 +277,21 @@ where
                     BeaconNodeHttpClient::new(url, Timeouts::set_all(CHECKPOINT_SYNC_HTTP_TIMEOUT));
                 let slots_per_epoch = TEthSpec::slots_per_epoch();
 
-                debug!(context.log(), "Downloading finalized block");
+                // We want to fetch deposit snapshot before fetching the finalized beacon state to
+                // ensure that the snapshot is not newer than the beacon state that satisfies the
+                // deposit finalization conditions
+                debug!(context.log(), "Downloading deposit snapshot");
+                let deposit_snapshot_result =
+                    remote.get_deposit_snapshot().await.map_err(|e| match e {
```

Comment:
Sorry to be a pain but maybe we should integrate this with the new `--disable-deposit-contract-sync` flag (https://github.com/sigp/lighthouse/pull/3597). Also happy to put it in an issue seeing as it's quite harmless.

Reply:
done!

Reply:
perfecto, let's merge??